external-crates/move/move-compiler/src/sui_mode/typing.rs

```diff
@@ -483,7 +493,14 @@ fn check_otw_type(
     }
 
     if let N::StructFields::Defined(fields) = &sdef.fields {
-        if let Some(invalid_field_loc) = invalid_otw_field_loc(fields) {
+        let InvalidOTWFieldLoc {
+            invalid_first_field,
+            more_than_one_field,
+        } = invalid_otw_field_loc(fields);
+        let mut check_field_loc = |invalid_field_loc_opt| {
+            let Some(invalid_field_loc) = invalid_field_loc_opt else {
+                return;
+            };
```

Comment:
As per my earlier comment, this would become (modulo fixes to nesting lower down):

```suggestion
        let Some(invalid_field_loc) = invalid_first_field.or(more_than_one_field) else {
            return;
        };
```